makes/binary.mk

```diff
@@ -28,24 +29,26 @@ LOCAL_HEADERS_DIR ?= include
 ABS_HEADERS_DIR := $(abspath $(LOCAL_DIR)/$(LOCAL_HEADERS_DIR))
 
 SRCS += $(addprefix $(LOCAL_DIR), $(LOCAL_SRCS))
+FILES := $(filter $(LANGUAGE_EXTENSIONS), $(SRCS))
```

Comment:
Why do we have to filter SRCS - are there any files other than *.c,*.cc, *.cpp inside? Do you have to introduce another variable? It's seems that SRCS is not used anywhere else so why not keep the result in SRCS?

Reply:
I thought it is a good idea to clean `SRCS` of any junk. Nevertheless, it could really be my over-thinking.

Reply:
Implicit `SRCS` filtering exists in previous versions of` binary.mk` and `static-lib.mk`, so I chose to keep them explicit. Nevertheless, I think it should have reused `SRCS` variable.

Reply:
OK, you mean it was filtered by patsubst earlier. I'm not sure about this filtering. What if someone makes typo in LOCAL_SRC for example `foo.ccc` instead of `foo.cc`. Will it go undetected?

Reply:
In previous version, all these files didn't change their extension to `.o` but patsubst continued to return them. In my version, they will remain undetected.

Reply:
in the end I left filtering